src/js/modal.js

```diff
@@ -1,156 +1,190 @@
-        const option = document.createElement('option');
+        const option = document.createElement("option");
         option.textContent = `${challenge.minParticipants + i} participants`;
         selectParticipants.appendChild(option);
     }
 
-    const button = document.createElement('button');
-    button.type = 'submit';
-    button.classList.add('button', 'primary');
-    button.name = 'submit-booking';
-    button.textContent = 'Submit booking';
+    const button = document.createElement("button");
+    button.type = "submit";
+    button.classList.add("button", "primary");
+    button.name = "submit-booking";
+    button.textContent = "Submit booking";
     form.appendChild(button);
 
-    form.addEventListener('submit', async e => {
+    form.addEventListener("submit", async (e) => {
         e.preventDefault();
 
-        await reserveTime(); 
-
-        renderContent(step3());
+        let requestedName = document.querySelector(".input-name").value;
+        let requestedEmail = document.querySelector(".input-email").value;
+        let requestedParticipants = document.querySelector(".select-participants").value;
+        let requestedTime = document.querySelector(".select-time").value;
+        let requestedDate = times.date;
+        let challengeId = challenge.id;
+        let matches = requestedParticipants.match(/(\d+)/);
+
+        
+
+        if (requestedName === "" || requestedEmail === "") {
+            alert("Please fill in requested fields.");
+        } else if (requestedParticipants === "-- Select number --") {
+            alert("Please select participants.");
+        } else if (requestedTime === "-- Choose time --") {
+            alert("Please choose time.");
+        } else {
+            requestedParticipants = parseInt(matches[0]);
+            await reserveTime(
+                challengeId,
+                requestedName,
+                requestedEmail,
+                requestedDate,
+                requestedTime,
+                requestedParticipants
+            );
+            renderContent(step3());
+        }
     });
 
     return form;
 }
 
 function step3() {
-    const div = document.createElement('div');
-    div.classList.add('modal-content', 'center');
+    const div = document.createElement("div");
+    div.classList.add("modal-content", "center");
 
-    const message = document.createElement('h1');
-    message.textContent = 'Thank you!';
+    const message = document.createElement("h1");
+    message.textContent = "Thank you!";
     div.appendChild(message);
 
-    const button = document.createElement('button');
-    button.classList.add('button', 'secondary');
-    button.textContent = 'Back to challenges';
-    button.addEventListener('click', removeModal);
+    const button = document.createElement("button");
+    button.classList.add("button", "secondary");
+    button.textContent = "Back to challenges";
+    button.addEventListener("click", removeModal);
```

Comment:
This doesn't actually change any logic, just the format of the code (switching between `"` and `'` quote marks).

Changes like these make a diff (the visualization of changes in a commit or PR) hard to follow. Avoid meaningless changes like these by agreeing on a format and sticking to it.